src/data/nat/fib.lean

```diff
@@ -96,4 +97,75 @@ begin
     mono }
 end
 
+/-- https://proofwiki.org/wiki/Consecutive_Fibonacci_Numbers_are_Coprime -/
+lemma fib_succ_coprime (n : ℕ) : gcd (fib n) (fib (n + 1)) = 1 :=
+begin
+  induction n with n ih,
+  { simp },
+  { convert ih using 1,
+    rw [fib_succ_succ, succ_eq_add_one, gcd_rec, add_mod_right, gcd_comm (fib n),
+      gcd_rec (fib (n + 1))], }
+end
+
+/-- https://proofwiki.org/wiki/Fibonacci_Number_in_terms_of_Smaller_Fibonacci_Numbers -/
+lemma fib_add (m n : ℕ) :
+  fib m * fib n + fib (m + 1) * fib (n + 1) = fib (m + n + 1) :=
+begin
+  induction n with n ih generalizing m,
+  { simp },
+  { intros,
+    specialize ih (m + 1),
+    rw [add_assoc m 1 n, add_comm 1 n] at ih,
+    simp only [fib_succ_succ, ← ih],
+    ring, }
+end
+
+
+lemma gcd_fib_add_self' (m n : ℕ) : gcd (fib m) (fib n) = gcd (fib m) (fib (n + m)) :=
+begin
+  cases eq_zero_or_pos n,
+  { rw h, simp },
+  replace h := nat.succ_pred_eq_of_pos h, rw [← h, succ_eq_add_one],
+  calc (fib m).gcd (fib (n.pred + 1)) 
+        = gcd (fib m) (fib (n.pred + 1) * fib (m + 1)) :
+    begin
+      have hcop := coprime.symm (fib_succ_coprime m),
```

Comment:
Is this assumption used?

Reply:
Used to close the goal with the next `rwa`.